mmv1/third_party/terraform/services/container/node_config.go.erb

```diff
@@ -1527,7 +1541,9 @@ func flattenSecondaryBootDisks(c []*container.SecondaryBootDisk) []map[string]in
 		for _, disk := range c {
 			secondaryBootDisk := map[string]interface{}{
 				"disk_image": disk.DiskImage,
-				"mode": disk.Mode,
+			}
+			if disk.Mode == "CONTAINER_IMAGE_CACHE" {
```

Comment:
It looks like this is an enum field with only one potential value; however, if new values are added to that field in the future (which happens from time to time) this will not work for those cases. What would you think about checking for empty or `MODE_UNSPECIFIED` here?

Reply:
Right now, if mode is not equal to CONTAINER_IMAGE_CACHE, gcloud automatically changes it into MODE_UNSPECIFIED. @elfinhe Do we want Terraform to reflect this behavior, or do we want it to reject modes that are not CONTAINER_IMAGE_CACHE?

Reply:
Discussing with @elfinhe, we want terraform to reject invalid modes instead of changing them to MODE_UNSPECIFIED. I've removed the mode changing logic, so the googleapi invalid mode error surfaces. This eliminates comparisons to CONTAINER_IMAGE_CACHE and associated logic. An example error is shown below with `mode="DATA" ` .

<img width="918" alt="Screenshot 2024-06-04 at 1 24 23 PM" src="https://github.com/GoogleCloudPlatform/magic-modules/assets/39973638/2e05e4d3-9146-4e5e-ba6b-30352cffa1ad">